src/main/java/seedu/address/logic/commands/DeleteTrainingCommand.java

```diff
@@ -66,6 +73,13 @@ private Student createEditedStudent(Student studentToEdit, Training trainingToDe
         return newStudent;
     }
 
+    private Student getStudentById(List<Student> studentsList, Id id) {
+        return studentsList.stream()
+                .filter(student -> student.getId().equals(id))
+                .findFirst()
+                .get();
```

Comment:
if studentsList empty, is it safe that .get() will not throw a null pointer error?

Reply:
It's assumed here that there will always be the Id existing in the list, is there a scenario that this wont happen?

Reply:
makes sense, but might be good to include an assertion here in case of code changes in future

Reply:
yep thanks, shall do that. We see how this find by id goes, best is we abstract it to a util method